evm_loader/lib/src/account_storage.rs

```diff
@@ -283,12 +284,42 @@ impl<'rpc, T: Rpc + BuildConfigSimulator> EmulatorAccountStorage<'rpc, T> {
         .await?;
 
         storage.download_accounts(accounts).await?;
+        let target_chain_id = tx_chain_id.unwrap_or(storage.default_chain_id());
+        storage.apply_balance_overrides(target_chain_id).await?;
```

Comment:
Why do you use overrides only in `EmulatorAccountStorage::with_accounts'? 
And not in other constructors?

Reply:
only this one is used in emulation calls, we discussed it with @s-medvedev, should I update others?

Reply:
updated others as well, please re-check, thanks!